Sources/RoutingKit/PathComponent.swift

```diff
@@ -1,5 +1,7 @@
 /// A single path component of a `Route`. An array of these components describes
 /// a route's path, including which parts are constant and which parts are dynamic.
+
+@dynamicMemberLookup
 public enum PathComponent: ExpressibleByStringLiteral, CustomStringConvertible {
```

Comment:
Maybe extending `PathComponent` would work?

```swift
extension PathComponent {
    @Parameter
    static var user: UUID
}
```

Reply:
I wish, and that was what I initially tried, but it turns out that you can't create key paths to static members. 😞

Reply:
Unless you have a completely different idea for the API, in which case I'm all ears.

Reply:
Ah I see. So you can't do dynamic member lookup statically. Dang. Maybe the lookup doesn't need to be dynamic member? Something like `req.parameters.get(\.$user)`?

Reply:
But that's still a key path... And if you extend `PathComponent` with properties with property wrappers they have to be static. If you want to use key paths the only option is to define your own type that you don't initialize and just use the stored properties for the key paths.

On the other hand, if you really want to extend `PathComponent`, you can have the `Parameter` property wrapper return a `PathComponent` instead of its type. If you did that the API would end up looking like this:

```swift
extension PathComponent {
    @Parameter<UUID>(name: "user_id") static var userID: PathComponent
}

router.get("users", .userID) { request in 
    return try request.parameters.get(PathComponent.$userID)
}
```

Reply:
Oh duh, I'm an idiot. Yeah that makes sense. Swift just needs to get static key paths.